### PR TITLE
New stock adjustment receipt

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR is related to #5458 by using just one document during an adjustment of stock, the part about the procedure is not touched

image

### CODE REVIEW

**Comment by mbayopanda** [comment]
The total cost of each line must be based on the difference and not on the quantity

File: server/controllers/stock/index.js
Comment (jniles) [inline server/controllers/stock/index.js]: Just FYI, you can do the `_.assign()` operation with vanilla javascript.  Like this:
```js
var x = _.assign({}, lot, {difference, is_exit: 0 });
// is equivalent to 
var y = { ...lot, difference, is_exit : 0 };
```

File: server/controllers/stock/reports/stock_adjustment.receipt.handlebars
Comment (jniles) [inline server/controllers/stock/reports/stock_adjustment.receipt.handlebars]: I think this `columns` needs to be 9.

File: server/controllers/stock/reports/stock_adjustment.receipt.handlebars
Comment (jniles) [inline server/controllers/stock/reports/stock_adjustment.receipt.handlebars]: I also think this is 9 columns.

File: server/models/schema.sql
Comment (jniles) [inline server/models/schema.sql]: Is there a particular reason you'd want to call this `log_timestamp`?  If not, let's call it the usual `created_at`, like we have in a couple other tables here.

File: server/models/schema.sql
Comment (jniles) [inline server/models/schema.sql]: Let's call this `data`, since it's JSON format.  Or `changes` or something more description.  We use `text` a lot for labels and this isn't a label.

File: server/controllers/stock/index.js
Comment (jniles) [reply server/controllers/stock/index.js]: I think this may end up being a lot of information to store in the JSON column.  Would it be possible to store only the `movement.uuid` and the `difference`?  Then, when you are rendering the receipt, you simply stitch them together with while displaying the receipt?

File: server/controllers/stock/index.js
Comment (jniles) [inline server/controllers/stock/index.js]: Cool 👍

File: server/controllers/stock/index.js
Comment (jniles) [inline server/controllers/stock/index.js:330]: It looks like there is a lot of shared/repeated information between these two `if`s.  Could you unify them a bit?

```js
const movementObject = {
          uuid : db.bid(uuid()),
          lot_uuid : db.bid(lot.uuid),
          depot_uuid : db.bid(movement.depot_uuid),
          document_uuid : db.bid(uniqueAdjustmentUuid),
          unit_cost : lot.unit_cost,
          quantity: Math.abs(quantity),
          date : new Date(movement.date),
          entity_uuid : movement.entity_uuid,
          flux_id : core.flux.INVENTORY_ADJUSTMENT,
          description : movement.description,
          user_id : req.session.user.id,
          period_id : periodId,
};

if (difference < 0) {
        countNeedDecrease++;
       Object.assign(movementObject, { is_exit : 1 });
       adjustmentLog.push(_.assign({}, lot, { difference, is_exit : 1 }));
} else {
        countNeedIncrease++;
       Object.assign(movementObject, { is_exit : 0 });
       adjustmentLog.push(_.assign({}, lot, { difference, is_exit : 0 }));
}

trx.addQuery('INSERT INTO stock_movement SET ?', movementObject);

```

You could actually probably get it smaller than this by adopting my proposal to remove `_.assign()`.  I think the only thing that changes is `is_exit`, right?

File: server/controllers/stock/index.js
Comment (jniles) [inline server/controllers/stock/index.js]: Do you need the user on both this table and the `stock_movement` table?  It seems like you probably only need the `document_uuid` and you could recover the user, right?

File: server/controllers/stock/index.js
Comment (jniles) [inline server/controllers/stock/index.js:366]: 😆

**Review by jniles — COMMENTED** [review]
@mbayopanda this is really cool!

I've made a few comments on the code and structure of the table.  I'm curious how much this may impact our other modules/registries, but it actually looks really clean.  Great job!

File: server/controllers/stock/index.js
Comment (jniles) [reply server/controllers/stock/index.js]: Actually, if you did that, you wouldn't need a JSON column.  Just two columns - `row_uuid` and `difference`.  You could then do the `JOIN` in MySQL!

**Comment by mbayopanda** [comment]
@jniles changes landed

File: server/controllers/stock/index.js
Comment (jniles) [inline server/controllers/stock/index.js:325]: Beautiful this is much cleaner.  :+1:

**Review by jniles — APPROVED** [review]
Code looks good to me.  Tested and works with the vanga dataset.

@mbayopanda I see you still have the "do not merge" flag on here.  Go ahead and merge this when ready.

bors delegate+

**Pull request merged**
